app/src/backend/GPCService.ts

```diff
@@ -0,0 +1,45 @@
+import { db } from "@/models";
+import createHttpError from "http-errors";
+
+type IDsFromReferenceNumberResult = {
+  sectorId: string;
+  subSectorId: string;
+  subCategoryId: string;
+};
+
+export default class GPCService {
+  public static async getIDsFromReferenceNumber(
+    gpcReferenceNumber: string,
+  ): Promise<IDsFromReferenceNumberResult> {
+    const subcategory = await db.models.SubCategory.findOne({
+      where: { referenceNumber: gpcReferenceNumber },
+      attributes: ["subcategoryId"],
+      include: [
+        {
+          model: db.models.SubSector,
+          as: "subSector",
+          attributes: ["subsectorId"],
+          include: [
+            {
+              model: db.models.Sector,
+              as: "sector",
+              attributes: ["sectorId"],
+            },
+          ],
+        },
+      ],
+    });
+
+    const subCategoryId = subcategory?.subcategoryId;
+    const subSectorId = subcategory?.subsector?.subsectorId;
+    const sectorId = subcategory?.subsector?.sector?.sectorId;
+
+    if (!sectorId || !subSectorId || !subCategoryId) {
+      throw new createHttpError.BadRequest(
+        "Couldn't find sector/ subsector/ subcategory for given GPC reference number",
+      );
+    }
+
+    return { sectorId, subSectorId, subCategoryId };
```

Comment:
One thing I'd want to check is for scope number (1, 2, 3). Sometimes the third element is the scope number; other times it's not. Worth including here!

Reply:
Fair, in case this method gets used in other places we can include the scope.

For the current use case we just need the three IDs to create an inventory value.

Since we need to do another join here to get the actual scope number from the Scope table, not sure if we need to do this now for an eventual use in the future...

Reply:
We could have a separate method GPCService.getScopeForReferenceNumber, since I don't think we use both our own IDs and GPC scope at the same time usually.